### DISCUSSION TITLE
Does CSharpDeclarationComparer and NamedTypeGenerator need to account for Records?

### DISCUSSION THREAD

**Discussion by Youssef1313** [dotnet/roslyn#47457] [General] [answered]
Currently, they don't handle records. I'm unsure if they need to handle records or not?

https://github.com/dotnet/roslyn/blob/b61d9d2863eebfc76b032e5f8fcd0a918fb27147/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpDeclarationComparer.cs

https://github.com/dotnet/roslyn/blob/a37aeb14c5608c3c213a6bc9ce0f9bc0b4c9fa10/src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

Also, CSharpCodeModelService

https://github.com/dotnet/roslyn/blob/30b18ba60e11d358425732e52ec3061deeb01105/src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.cs#L856

**Comment by jmarolf** [comment, marked as answer]
Curious what @CyrusNajmabadi's opinion on this is. My assumption is that we need to update all our internal APIs to support records as a target

**Reply by CyrusNajmabadi** [reply]
yes.  i think these would likely need to be updated.

**Reply by Youssef1313** [reply]
@CyrusNajmabadi Can you point me to where the existing tests for this to extend them for records?
